### PR TITLE
tweak issue template

### PR DESCRIPTION
**Pull request by SethTisue**
* ask for Scala version
* tweak the "reproduction steps" to encourage including both the code and the results, and to subtly encourage a REPL-based reproduction if one exists
* collapse last two sections, I've never understood what the difference between them is supposed to be, and based on the bug reports that have come in in the last couple months, I think others don't either

### CODE REVIEW

**Comment by SethTisue** [comment]
an example of the last two sections being separate leading to annoying and useless repetition in reports is https://github.com/scala/bug/issues/11982, another is https://github.com/scala/bug/issues/11984, another example is https://github.com/scala/bug/issues/11976 , and those are literally the first three tickets I looked at

**Comment by SethTisue** [comment]
re: Scala version, I'm a bit tempted to also add “is the problem reproducible in the latest Scala release?", but I'm trying to stay away from the include-just-one-more-bit-of-advice slippery slope

**Comment by dwijnand** [comment]
> * collapse last two sections, I've never understood what the difference between them is supposed to be, and based on the bug reports that have come in in the last couple months, I think others don't either

> an example of the last two sections being separate leading to annoying and useless repetition in reports is #11982, another is #11984, another example is #11976 , and those are literally the first three tickets I looked at (!)

Still very much disagree.  It's an (eagerly evaluated, rather than maintainer-remembered, prompted, and answered) opportunity to try to explicitly record more details about the reporter's expectations.  "We expect to load the Test" - guessable from the REPL session paste, but good to know for sure that's the user's expectation.  "I expect the behavior of LazyList to be at least as lazy as Stream." - great, "at least as lazy as", good generalisation to bear in mind from the user's report.

> re: Scala version, I'm tempted to also add “is the problem reproducible in the latest Scala release?", but I'm trying to stay well away from the edge of the include-just-one-more-bit-of-advice slippery slope

Yeah, I feel like that's one question too far.

I'm happy to add the Scala version request, but if it were that or the problem vs expectation split, I'd keep the latter.  What's the motivation for adding the version?  Asking because I haven't seen it as a (meta 🙂) issue in the reports.

**Comment by dwijnand** [comment]
But I'm not overly fussed if you feel strongly enough about this.

**Comment by SethTisue** [comment]
@dwijnand can you show me a good example of when the last two sections being separate resulted in a better bug report? because offhand I don't think I've seen one, whereas the counterexamples just keep coming

perhaps with a good example in hand, I could craft a wording that would satisfy me as much but dissatisfy you less :-)

> What's the motivation for adding the version? Asking because I haven't seen it as a (meta 🙂) issue in the reports.

(I'm not super attached to this change, but I am in favor, because:)

I have seen a few reports where we found out that someone wasn't running the latest patch version. so, visible meta-issue from time-to-time

But more often it's an invisible meta-issue: when people don't include the Scala version, literally my _first_ question about nearly _any_ bug report is, were you on 2.11 or 2.12 or 2.13, and were you on the latest patch version? even if there's nice reproduction steps, the _first_ thing I do is paste it into a 2.13.2 REPL to see if it's even still reproducible

**Comment by eed3si9n** [comment]
The expectation section becomes relevant when the behavior is not a simple programming mistake, but more of a difference in intention, design, and/or the spec interpretation.

Having the problem and expectation spelled out I think would make the report clearer, and we can have a conversation about whether we agree on the expectation before we jump into how that may be achieved.

For example, here a recent sbt issue https://github.com/sbt/sbt/issues/5520, whose expectation is "All dependencies are downloaded like Maven does." Coursier found inconsistency in checksum and it threw an error apparently. My expectation does not match with the OP, and it's good to know that upfront.

Here's another one - https://github.com/sbt/sbt/issues/5523. I have been able to guess what the expectation was from the steps, but it'd be difficult because the error message is sbt displaying  a parsing error. The expectation clarified OP's intent.

So maybe we can find better examples in scala/bugs from closed issues? See for example https://github.com/scala/bug/issues/11961. Here's another closed one https://github.com/scala/bug/issues/11953.

**Comment by SethTisue** [comment]
> Having the problem and expectation spelled out I think would make the report clearer

in theory tat sounds fine, but in practice I have no idea what "problem" and "expectation" are even supposed to mean, in opposition to each other, and I think our experience has been that the people filing bug reports don't know that, either

I don't think https://github.com/sbt/sbt/issues/5520 helps make your case, because the "expectation" section merely repeats what they already said under "problem", which is exactly what I'm seeing over and over again in scala/bug

https://github.com/sbt/sbt/issues/5523 doesn't help make your case, it helps make mine — OP filled in "I could only guess.." under Problem because they felt they had to write something, so they had to spend time coming up with some useless verbiage to fill in, and we had to spend our time reading it

similarly, in https://github.com/scala/bug/issues/11953, the "expectation" uselessly repeats the problem, extra words that add nothing

https://github.com/scala/bug/issues/11961 is a nice report, but I think that's because the reporter was thoughtful; I think they'd have been equally thoughtful with my proposed template revisions

I'm not convinced by any of this, if anything it solidifies my position.

**Comment by eed3si9n** [comment]
> in theory that sounds fine, the intention is good, but in practice I have no idea what "problem" and "expectation" are even supposed to mean, in opposition to each other, and I think our experience has been that the people filing bug reports obviously don't know that, either

ok. Maybe that needs clarification.

> ## problem
> 
"problem" is intended to be an observation / objective description of the current Scala behavior that the OP finds it to be incorrect. 
> 
> ```scala
> scala> "1" == Option("1")
> res0: Boolean = false
> ```
> 
> The expression returns `false`.
> 
> ## expectation
> 
I think the natural response to the problem is "so what did you expect to happen instead?" In case of an error message it might be "no error". The expectation is my subjective judgement on what _should_ happen instead, which could be different from yours.
> 
> I expect the code to not compile.
> 
> ## notes
> 
notes is where I could put on the programmer hat and try to guess why it's happening, or link to previous discussions for or against my expectation.

In other words, the problem section is about "can I reproduce your problem?" and there shouldn't be difference in opinion.

Whereas expectation could be more nuanced. Like given the existing code base we can't make this semantic changes etc.

**Comment by lrytz** [comment]
Just my vote: I share Seth's views on this (after reading the good arguments and observations shared here).

**Comment by som-snytt** [comment]
The anti-templaters want to minimize the template, like Republicans seek to reduce the size of government by destroying it.

Compare doti's "minimized code" and "output", which is redundant for REPL session.

I think "copes well with ambiguity" is a good low bar for submitting an issue.

A drop down or checkboxes for Scala version would be great! Totally agree it's my first question!

Tweak the last section to say **expectoration**.

**Comment by som-snytt** [comment]
Example where my expectation remains completely at odds with the implementation: https://github.com/lampepfl/dotty/issues/8092

**Comment by eed3si9n** [comment]
> The anti-templaters want to minimize the template, like ...

I really don't like this ad hominem here. I have no doubt that both Seth and Lukas are making their points in good faith of what makes sense for all of us to keep track of the bugs in scala/scala.

**Comment by som-snytt** [comment]
@eed3si9n Sorry, I hoped the tone of levity came though in every word of that comment.

I meant to add, "(_before_ reading the good arguments and observations shared here)", except it would not have the whiff of truth that garnishes humor.
